### ISSUE TITLE
Warcraft III: staying in menus quickly gobs all system memory

### ISSUE THREAD

**Issue by sim590** [doitsujin/dxvk#1866]
In a matter of at most 20 seconds of staying in the menus, dxvk will reserve all of my 32 GBs or RAM. Starting an game stops the leak though. But as soon as I get back in the menus after a game finishes, the memory leaks accumulate again.

Uninstalling dxvk will remove the memory leak problem and the same attempt at reproducing the issue is not successful. Therefore, dxvk must be leaking.

> make sure to test with latest `master` as well.

I did try and the same happens. The bug is not solved in `master`.

### Software information

The game is Warcraft III Reforged.

### System information
- GPU: Radeon RX 5700XT
- Driver: amdgpu
- Wine version: wine-5.22 (Staging)
- DXVK version: 1.7.3

### Log files

[Warcraft III_d3d11.log](https://github.com/doitsujin/dxvk/files/5742069/Warcraft.III_d3d11.log)
[Warcraft III_dxgi.log](https://github.com/doitsujin/dxvk/files/5742070/Warcraft.III_dxgi.log)
[wine.log](https://github.com/doitsujin/dxvk/files/5742089/wine.log)

**Comment by doitsujin** [comment]
apitrace?

**Comment by sim590** [comment]
> apitrace? 

I can't manage to make one. When trying the 64-bit release of apitrace, I'm getting:

```
error: binaries mismatch: you need to use the 32-bits apitrace binaries to trace this application
```

and when trying with the 32-bit libraries, the program doesn't load. I'm left with that output that doesn't stop:

[wine.log](https://github.com/doitsujin/dxvk/files/5742748/wine.log)

The thing is that any executable shipped by Blizzard will lead to start Battle.Net and require the user to start the game from the Battle.Net app. The Battle.Net app is 32-bit while the Warcraft III reforged executable is 64-bit. Therefore, I think that I can't start the apitrace in that configuration. Actually, I have two choices for starting Warcraft III:

```
Warcraft III Launcher.exe: PE32 executable (GUI) Intel 80386, for MS Window
```
and
```
_retail_/x86_64/Warcraft III.exe: PE32+ executable (GUI) x86-64, for MS Windows
```

The first will launch Battle.Net directly. I will get the first message about using 32-bit apitrace if I'm using 64-bit and the program not loading when trying 32-bit. As for the second choice of executable, when trying to run with 64-bit, the program will start Battle.Net, but I don't see any trace written anywhere. Here is what I entered:

```
$ env TRACE_FILE=apitrace.log DXVK_HUD=memory WINEPREFIX=/var/remise/WineBattleNet wine /home/simon/Téléchargements/apitrace-latest-win64/bin/apitrace.exe trace -a dxgi -v _retail_/x86_64/Warcraft\ III.exe 
```

There is no file apitrace.log in any subdirectory. If you have any suggestion, I'm ready to follow your instructions.

> Also, does the leak show up with `DXVK_HUD=memory`?

No. It seems like it is the `BlizzardBrowser` process that takes the most memory according to `htop`. However, the 3% doesn't match the amount that is taken by the memory bar at the top. Here I stopped at 17GB of memory.

![notshown](https://user-images.githubusercontent.com/4147254/103146051-a6008480-4711-11eb-9726-a03250c122d4.png)

>  and Warcraft III is fairly old so I'd somehow expect people to have run into this before...

The game is Warcraft III Reforged. It has been "remastered" recently in the last year.

### Addtional logs

I have also found those logs:

[BlizzardBrowser_d3d11.log](https://github.com/doitsujin/dxvk/files/5742758/BlizzardBrowser_d3d11.log)
[BlizzardBrowser_dxgi.log](https://github.com/doitsujin/dxvk/files/5742759/BlizzardBrowser_dxgi.log)

I think that BlizzardBrowser is the program that executes when running the game menu. It's a web based app that launches the game when you start an actual Warcraft game.

**Comment by doitsujin** [comment]
> Therefore, I think that I can't start the apitrace in that configuration.

You can just copy the correct d3d11.dll, dxgi.dll and dxgitrace.dll next to the game's executable.

**Comment by sim590** [comment]
I managed to make spawn an `apitrace.log` file. The size is 3.38GB... I assume I can't upload it here. I did upload it [here](https://mega.nz/file/6jBW2DqL#Nts_gyRLlVTG63IdEwWyoTPcyW3WxH1IDQiTUvpExiU).

**Comment by K0bin** [comment]
Not sure why you renamed it to 'apitrace.log'. Apitrace would've created a file called 'Warcraft.III.trace'.

Anyway, unfortunately the file is broken and ends after just 14 D3D API calls:
![grafik](https://user-images.githubusercontent.com/1131720/103152128-23191180-4785-11eb-9a6b-8c76ce63d5d8.png)

**Comment by sim590** [comment]
@K0bin is it due to the fact that I did `kill -9` the process? I would assume so. I will try to make another one but ending the program sooner before I'm forced to kill it.

**Comment by sim590** [comment]
I have regenerated another apitrace. Don't know if it's good though...

[apitrace.log](https://github.com/doitsujin/dxvk/files/5762777/apitrace.log)

**Comment by sim590** [comment]
The file is [here](https://mega.nz/file/inxWGLjA#m0LH1qRSbPdYgC4Kzd_srNF-FIpDOh2FOAKkGmb36Nc).

**Comment by stanczew** [comment]
You can launch the game directly, skipping Battle.net App, by going to `Warcraft III/_retail_/x86_64` and running:
```
wine "Warcraft III.exe" -launch
```
This will take you to the login screen inside the game (since login info is not provided by Battle.net App), where the memory leak can already be observed. For me with 16 GB of RAM it's a matter of 5-10 seconds before OOM killer is invoked.

> If the problem isn't the game itself but rather the launcher, just setting DLL overrides for the launcher itself (d3d11 and dxgi to `builtin`) should work around the problem.

It's neither the game nor the launcher, but I can confirm this workaround: setting d3d11 and dxgi to builtin **just for BlizzardBrowser.exe** makes the issue go away.

Also I gathered an [apitrace](https://drive.google.com/file/d/1Z-o4NeriYbFi3UBDqReV-ccvg8wFKntM/view) for BlizzardBrowser. It's with DXVK enabled, since with WineD3D I get an exception when launching the game.

**Comment by doitsujin** [comment]
What was the trace recorded with? I'm just getting `error: () unknown call detail 32` when trying to open it.

**Comment by stanczew** [comment]
I followed the instructions in [the Wiki](https://github.com/Joshua-Ashton/d9vk/wiki/Making-a-Trace), and copied the four dlls into `BlizzardBrowser` directory (which is one level below the directory of `Warcraft III.exe`). The only exception is that I didn't disable DXVK, because then 1) the leak is not reproducible, and 2) I couldn't start the actual game.

I then launched Warcraft III.exe as described above. I got BlizzardBrowser.trace files (there are multiple BlizzardBrowser processes spawned when launching the game), and I also got apitrace logs in the Wine output, so I thought it had worked (I've never made an apitrace before).

My system info:
- GPU: Radeon RX 570
- Driver: amdgpu
- Wine version: 6.1 Staging, esync enabled
- DXVK version: 1.7.3

**Comment by doitsujin** [comment]
can you open your own trace file with qapitrace?

**Comment by stanczew** [comment]
Nope, I get the same error as you.

I'm not sure how/if I can get a useful trace. I see in the new issue template that for D3D11, instead of copying apitrace wrapper dlls, I should run `apitrace.exe` on the target application. But, I'm not manually launching `BlizzardBrowser.exe`; and if I run apitrace on `Warcraft III.exe` I get a bunch of `apitrace: warning: caught exception 0xc000001d` logs in the console, and the resulting trace is just 2.5 KiB / 5 calls.

Then, according to [this comment](https://github.com/doitsujin/dxvk/issues/1808#issuecomment-722698595), it should nevertheless be possible to use the "copy dlls" method; but no matter how I try to do this, I still get `unknown call detail <number>` when viewing the trace.

**Comment by sim590** [comment]
@stanczew :

> It's neither the game nor the launcher, but I can confirm this workaround: setting d3d11 and dxgi to builtin just for BlizzardBrowser.exe makes the issue go away.

Can you explicit how you go about overriding the DLL just for `BlizzardBrowser.exe` and not for the game? The game actually needs native DLLs for dxvk to make its magic and make the game playable. Otherwise, the game graphics are really messed up with low gamma.

**Comment by stanczew** [comment]
> Can you explicit how you go about overriding the DLL just for `BlizzardBrowser.exe` and not for the game?

Run `winecfg`, on the "Applications" tab click "Add application…" and choose `BlizzardBrowser.exe`, click on it so it's focused, then go to "Libraries" tab and add d3d11 and dxgi as `builtin`:
![image](https://user-images.githubusercontent.com/29931025/107144942-b32ea700-693e-11eb-9ced-6e1513b78089.png)
![image](https://user-images.githubusercontent.com/29931025/107144962-c8a3d100-693e-11eb-940c-edc03d2dff26.png)

**Comment by cg666** [comment]
I have the same issue. I enabled DXVK_HUD=memory and I don't see the game eating memory from HUD infos nor with "top" command.

But, the GPU memory get filled very quickly. I set up the OoM Killer and it is not triggered. I suspect memory allocated on GPU and never released. That's why the OoM Killer does nothing.

I have NVIDIA GTX 960. It seems not related to GPU driver as other reports issue on AMD.

Hope it can help.

**Comment by Maia-Everett** [comment]
Thanks, stanczew, the workaround seems to have worked for me. BlizzardBrowser still leaks memory a bit, but at such a slow pace that it's going to be unnoticeable in practice, since you aren't going to stay in the main menu for that long.

**Comment by kostile** [comment]
I am attempting to play WC3 as well but can't do so currently with the instructions here. The exact same thing happens no matter what if I override d3d11 and dxgi with. I should also note that this only happens on wine version 6.0+ and doesn't seem to happen on earlier versions. However, I need to play with this version as it's the only one supporting avahi/bonjour properly at this time.

**Comment by HolimaX** [comment]
Workaround gives just a little extra time to start Campaign. It makes game navigable, but not really enjoyable.. just yet.

**Comment by nerzhul** [comment]
hello,
i have the issue (tried workarounds) starting when i enter in a custom map multiplayer party, the memory fills in 10sec and bye :(

**Comment by ab0v3g4me** [comment]
I also get huge memory allocations when running the game and it runs nice for like 10 seconds and then just freezes the game.

Overriding the `BlizzardBrowser.exe` didn't do much.

**Comment by RadicalGDPRist** [comment]
Same issue, the workaround from above works. I could not get a useful trace on Windows 7 x64 either. Most of the 67MB trace are calls to "ID3D11Device1::CheckMultisampleQualityLevels" and then basically The End.

I'm ready to provide the game files, I don't think it needs an account/installation for repro as you can access the main menu; or my account if the dev is willing to test.

# Detailed information:

Warcraft 3 Reforged uses CEF/Chromium v69 for all the new menu parts. This includes: main menu, map loading screen, settings screen. The old game renderer was left untouched, I think it kind of looks like this:

                        ↙ Legacy game engine renderer (dx9/dx11/opengl)
    Presentation/Compositor, reported as dx11
                        ↖ CEF menu renderer (apparently dx11?)

Game's launch options `-graphicsapi opengl/direct3d11/direct3d9` did not change Warcraft 3.exe's reported used API, but dx9 has 3x performance improvement with in-game renderer for me on Windows, so _that part_ works. The in-game renderer works flawlessly under Wine.

**The CEF renderer leaks ~1GiB of memory every time you hover over all 6 main menu buttons** ([see on the left](https://github.com/doitsujin/dxvk/issues/1866#issuecomment-751325314) behind htop window). Basically it leaks memory any time a menu element is re-rendered. You can get quite far with 32GB RAM and 8GB swap, but you'll freeze/OOM eventually. htop does not show an increase in res/virt memory for the BlizzardBrowser.exe process. This excessive garbage creation explains the low menu FPS too (buffer not reused?).

For my tests the main menu background animation removed and not rendered (see "Allow Local Files", the video was replaced with an empty "Warcraft III/_retail_/webui/webms/mainmenu.webm"). There was no increase in used memory when I did not move the cursor.

PS: Trying to execute apitrace the way it's meant to be prints: "error: failed to execute inject". Putting all three .dll's (d3d11.dll, dxgi.dll, dxgitrace.dll) prints: "error: unavailable function CreateDXGIFactory2" and exits the game process (this method was suggested above). Removing dxgi.dll floods a lot of "apitrace: warning: caught exception 0xc000001d", "0xc0000420", "0x80000003" etc. but produces no useful trace (only CheckMultisampleQualityLevels). I'm not sure how to do it right.

**Comment by epatpol** [comment]
I've tried the above workarounds (overriding d3d11 and dxgi in wine configuration) and I still have horrible memory leaks where even with 32gb ram I'll crash before entering a custom or a normal game. Anybody made some progress with this?

**Comment by nerzhul** [comment]
@epatpol tried 1 week ago, same issue yes

**Comment by Arnoldi** [comment]
I have the same problem. I had the game running for more than a year, but recently deleted and reinstalled the game (nothing else in my system). I lost the config I had, but now I can't get it to not leak memory anymore.

**Comment by epatpol** [comment]
@Arnoldi Yea same for me, I tried setting older versions of wine or dxvk in lutris but it's always the same, as soon as I arrive in the menu my memory consumption ramps up so quickly I have to find a game within a minute or so. It's basically unplayable for me right now. I have no idea what the culprit is however, does it leak in Windows as well but maybe at a slower rate? I've seen wc3 streamers having to restart their wc3 every couple of hours because they say it ends up crashing if you let it running...

**Comment by ChristophHaag** [comment]
I've had this too. A few observations:

* It usually starts happening when upgrading wine (I use lutris and the lutris builds)
* It randomly stops happening after starting warcraft with old versions of mesa, but not always. pattern very unclear. mesa shader cache doesn't seem to be related either.

It's definitively BlizzardBrowser eating VRAM, I would assume it's a bug in the texture sharing between libcef and warcraft too. The rate at which the VRAM fills is directly related to the resolution of the warcraft window. You can watch the VRAM being eaten live e.g. with `radeontop`. Once the VRAM is full, it starts eating GTT, which is system ram that is provided to applications instead of physical VRAM.

On AMD you can `cat /sys/kernel/debug/dri/0/amdgpu_gem_info` as root to see who allocates how much VRAM. At 4k BlizzardBrowser makes lots of 134 mb allocations

    pid  2104851 command BlizzardBrowser:
    [...]
		0x00000070:    134217728 byte VRAM NO_CPU_ACCESS EXPLICIT_SYNC
		0x00000071:    134217728 byte VRAM NO_CPU_ACCESS EXPLICIT_SYNC
		0x00000072:    134217728 byte VRAM NO_CPU_ACCESS EXPLICIT_SYNC
		0x00000073:    134217728 byte VRAM NO_CPU_ACCESS EXPLICIT_SYNC
		0x00000074:    134217728 byte VRAM NO_CPU_ACCESS EXPLICIT_SYNC
    [...]

**Comment by krablord** [comment]
Played without any issues pretty much on release. Right now it memory leaks all the way to game freeze (vram and ram), none of the workarounds helped. 
Ubuntu 20.04 + Nvidia 470

**Comment by xadips** [comment]
Indeed a few months ago I could still play normally, right now I'm almost always getting memory leaks on menu and it fills up my 32 GB ram so quickly, none of the workarounds worked, tried different wine versions/settings, sometimes it does not leak on wined3d

**Comment by Lufen34** [comment]
I'm getting the same issue. 64 Gb of Ram taken in less than 20 seconds. :disappointed: 

All above workaround did not work for me :|

Fedora 35 + Nvidia 510.60.02

**Comment by krablord** [comment]
With new PTR it never loads in to menu properly, but don't eat up any memory :/

**Comment by Blisto91** [comment]
@krablord If you are able then please post dxvk logs and if possible a apitrace from windows or wined3d if any of them work properly.

**Comment by krablord** [comment]
@NikiSkaarup sadly didn't work for me - maybe u can specify your settings like wine
@Blisto91 - can't find dxvk log anywhere for wc3, where should I look? (not in working (prefix) folder)

**Comment by Blisto91** [comment]
@krablord It should be in the games folder. Usually next to the exe or the top folder.
If you are running it through lutris you will either need to enable logging/debugging or launch with `DXVK_LOG_LEVEL=info` since the lutris wine runner disables the logging by default.
You can also speificy a specific path with `DXVK_LOG_PATH=`

**Comment by krablord** [comment]
@Blisto91 enabling debugging in Lutris only thing that worked to create any logs :/, these are 3 with anything inside
[BlizzardBrowser_d3d11.log](https://github.com/doitsujin/dxvk/files/8921965/BlizzardBrowser_d3d11.log)
[Warcraft III_d3d11.log](https://github.com/doitsujin/dxvk/files/8921966/Warcraft.III_d3d11.log)
[Warcraft III_dxgi.log](https://github.com/doitsujin/dxvk/files/8921967/Warcraft.III_dxgi.log)

**Comment by NikiSkaarup** [comment]
@krablord I've not really changed much, but it might be a combo of the wine version `lutris-GE-Proton7.16-x86_64` I'm using with DXVK v1.10.1, that somehow works

![Screenshot_6](https://user-images.githubusercontent.com/7352837/174348865-cf0cea41-3629-497c-9799-0d4e8cdaddf8.png)
![Screenshot_5](https://user-images.githubusercontent.com/7352837/174348889-41983498-01fa-415d-abc5-f2318edaea07.png)

**Comment by ElusiveMori** [comment]
Did some digging around the PTR issue.

Tried:
* DXVK with Proton Experimental with steam
* DXVK with wine-tkg
* Whatever the heck else I don't remember

Vanilla wine does not work because of missing Media Foundation libs - didn't bother.

These issues are unique to the PTR. Things work fine on the old patch with either config.

## Proton progress
CEF refuses to load with:
```
[0623/222921.947:ERROR:network_change_notifier_win.cc(142)] WSALookupServiceBegin failed with: 0
[0623/222922.102:WARNING:gpu_process_host.cc(1217)] The GPU process has crashed 1 time(s)
[0623/222922.251:WARNING:gpu_process_host.cc(1217)] The GPU process has crashed 2 time(s)
[0623/222922.401:WARNING:gpu_process_host.cc(1217)] The GPU process has crashed 3 time(s)
[0623/222922.550:WARNING:gpu_process_host.cc(1217)] The GPU process has crashed 4 time(s)
[0623/222922.699:WARNING:gpu_process_host.cc(1217)] The GPU process has crashed 5 time(s)
[0623/222922.848:WARNING:gpu_process_host.cc(1217)] The GPU process has crashed 6 time(s)
[0623/222922.848:ERROR:browser_gpu_channel_host_factory.cc(153)] Failed to launch GPU process.
[0623/222923.250:ERROR:network_change_notifier_win.cc(142)] WSALookupServiceBegin failed with: 0
```

Apitrace for BlizzardBrowser:
![image](https://user-images.githubusercontent.com/3159552/175390118-ca7c89fd-22ac-4a9c-b3b1-456af7362d19.png)

The menu never boots up as a progress. CEF just gives up and hangs.

Wine or DXVK logs don't show anything particularly interesting or alarming. [Full info with trace file and logs attached](https://github.com/doitsujin/dxvk/files/8970671/proton-experimental.tar.gz)

## Wine-tkg progress
BlizzardBrowser refuses to load with:
```
[INFO]-[BNL_Browser]: 2022-06-23T22:32:35 {480} [INFO] - setting background color '0'.
[INFO]-[BNL_Browser]: 2022-06-23T22:32:35 {480} [INFO] - UserAgent: BlizzardBrowser/5.2.3.579 Chrome/83.0.4103.106
[ERROR]-[BNL_Browser]: 2022-06-23T22:32:35 {480} [ERROR] - Failed to find the GPU adapter with the specified Luid.
[ERROR]-[BNL_Browser]: 2022-06-23T22:32:35 {480} [ERROR] - failed to initialize shared texture factory.
[ERROR]-[FileIO]: 2022-06-23T22:32:35 {480} [ERROR] - Removing READ_ONLY attribute failed with error code: 32
```

apitrace suggests it doesn't like something about the device:
![image](https://user-images.githubusercontent.com/3159552/175392359-203af502-961c-4c3f-b01e-bc3fd3412b13.png)

Otherwise, nothing particularly interesting or alarming [in the logs](https://github.com/doitsujin/dxvk/files/8970693/wine-tkg.tar.gz)

## Thoughts

In some other configurations

**Comment by nvaert1986** [comment]
I can confirm that the workaround works for the stable version.

I can also confirm that version 1.33.0 PTR version does not work with the latest wine-staging-7.13 and dxvk-1.10.2.  The game boots, shows the Blizzard logo movie, shows the loading background and then shows a black screen. The sound and music in the background do seem to be playing.

**Comment by Shin-Aska** [comment]
> @krablord I've not really changed much, but it might be a combo of the wine version `lutris-GE-Proton7.16-x86_64` I'm using with DXVK v1.10.1, that somehow works
> 
> ![Screenshot_6](https://user-images.githubusercontent.com/7352837/174348865-cf0cea41-3629-497c-9799-0d4e8cdaddf8.png) ![Screenshot_5](https://user-images.githubusercontent.com/7352837/174348889-41983498-01fa-415d-abc5-f2318edaea07.png)

Thanks, I only needed to add STAGING_SHARED_MEMORY=1 in my environment variables and it now did not gobble up memory on the main menu (on the non-ptr/main release)

**Comment by nvaert1986** [comment]
Are we sure Vulkan Shared Memory is not supported? AS there was a patch for this in 2019: https://www.phoronix.com/news/Wine-Vulkan-Shared-Resources

**Comment by Blisto91** [comment]
If you mean dx & vulkan shared resources then that has just been added. 
The wine patches are available as a [option in tkg now](https://github.com/Frogging-Family/wine-tkg-git/blob/master/wine-tkg-git/customization.cfg#L102) and should also appear in proton experimental before long. They will need to be used with dxvk 1.10.3 or above (master)

**Comment by ElusiveMori** [comment]
This is great news. Will report on my findings when I get around to it.

**Comment by kolAflash** [comment]
Blizzard released the update 1.32.x -> 1.33.0.19194 last night (2022-08-18).
Since then the main menu is just black. (at least with my Radeon RX 590)

That problem is also happening without DXVK. So I opened a bug at Wine.
https://bugs.winehq.org/show_bug.cgi?id=53573

**Comment by Blisto91** [comment]
Might be a separate issue.
Could you try again with Proton Experimental Bleeding Edge? The required stuff hadn't merged last time i posted here.

**Comment by ElusiveMori** [comment]
@kolAflash Same issue as on PTR. Need proper shared resource support in both wine and dxvk for it to work again.

**Comment by kolAflash** [comment]
What does "PTR" mean in this context?

@ElusiveMori  
So you can reproduce?  
Which kind of gpu do you use? (Radeon!?)
Or are you sure this may not be something gpu driver specific?

**Comment by ElusiveMori** [comment]
@kolAflash 
PTR stands for "Public Test Realm". It's basically the beta branch of WC3. Look up my previous explainers on the issue. 

tl;dr it was pretty much a guarantee that 1.33 would break linux compat when the PTR went live. It is caused by them switching the way they render their web ui, which isn't (fully) supported yet under Wine & DXVK

**Comment by kolAflash** [comment]
@ElusiveMori  
Aww crap. Now I also get the discussion about the [PTR](https://us.forums.blizzard.com/en/warcraft3/t/how-to-access-the-warcraft-iii-reforged-ptr/24388) for [1.33](https://us.forums.blizzard.com/en/warcraft3/t/warcraft-iii-version-133-public-test-realm-is-now-available/27992) that went on here before. Thanks 👍
So what I reported is probably [exactly that](https://github.com/doitsujin/dxvk/issues/1866#issuecomment-1156233610). And that PTR code just went live when WC3 Reforged 1.33 was released last night.

**Comment by SeanMcnabb** [comment]
Even when compiling with that patch, I'm getting this when attempting to run WC3. I am on an AMD graphics card if that makes a difference.

![image](https://user-images.githubusercontent.com/25807635/185524195-92c48484-6c25-4d1f-b312-964950481342.png)

**Comment by Blisto91** [comment]
Glad to hear this might seem to be working now. 🙂 

This version of the shared resources stuff probably won't make it into wine staging. The guy who created it is not at the moment pursuing getting it upstreamed since he says it's "kindof a hack" and would rather have a better solution. Afaik at least, I'm not involved myself 😁

**Comment by ElusiveMori** [comment]
Can confirm. Running on `tkg-7.15.r4` works when set to Win 7 compat mode.

**Comment by DistantThunder** [comment]
Can't make this work somehow, I made a local build like mentioned ( exact same commit in addition, set the custom build flags) and used DXVK 1.10.3 but I get either black screen or stuck on lion's head, Lutris or no Lutris. Weird...

**Comment by Blisto91** [comment]
Try the new wine-ge release togethet with lutris. It should have it all included plus stuff like fshack etc.
https://github.com/GloriousEggroll/wine-ge-custom/releases/tag/GE-Proton7-26

**Comment by DistantThunder** [comment]
I can confirm Wine-GE-Proton7-26 fixes this for me. Thanks for the heads up @Blisto91. I'm not seeing the telltale Memleak.

**Comment by Blisto91** [comment]
Glad you got it working.

If any of you find any new issues you think dxvk is the cause then i encourage you to make a new issue. 🙂

**Comment by orklann** [comment]
@Blisto91 I download Wine-GE-Proton7-26 with lutris, and it still shows me a black screen.

**Comment by orklann** [comment]
Do I need to set to Win 7 compat mode? I tried to set to Win7 compat mode, but battle net client can not opened.

**Comment by orklann** [comment]
Okay, I need to do everything from scratch, will upload the log later.

**Comment by orklann** [comment]
@Blisto91 It seems there is no log appear next to my .exe file.

**Comment by Blisto91** [comment]
Even after enabling Output debugging info for the games runner?

**Comment by orklann** [comment]
Yes, I have enabled Output debugging info in the games runner at the bottom of the window.

**Comment by Blisto91** [comment]
In the lutris environment variables can you add `DXVK_HUD` and then set it to `1` and then verify the hud shows in the top left corner ingame

**Comment by orklann** [comment]
Thank you for your tips, I will it later. Now i am building wine-tkg from what @kolAflash  suggest, and install dxvk 1.10.3 by winetricks.

**Comment by Blisto91** [comment]
Roger. This should hopefully start working better out of the box when when more custom wine versions start including this out of the box. E.g lutris and bottles own wine.

A separate thing to try would be to add it as a non steam game in steam and then force it through proton experimental.

**Comment by orklann** [comment]
@Biswa96 I tried that yesterday, with proton experimental it still the same issue.

**Comment by orklann** [comment]
@Blisto91 Do you mean the non-steam games with proton experimental setup?

**Comment by orklann** [comment]
No, I have removed it.

**Comment by Blisto91** [comment]
k.

I kinda forgot to ask. What is your gpu and drivers.

**Comment by orklann** [comment]
Mesa Intel® Iris(R) Plus Graphics 640 (Kaby Lake GT3e) (KBL GT3)

Ubuntu 22.04 with default GPU driver.

**Comment by Blisto91** [comment]
Okay that could maybe be part of the cause. I don't have any experience myself with intel iGPU's but i think don't they have as well of a track record as amd and nvidia with dxvk. At least not the older ones.
I would very much advice making sure you have the latest stable drivers you can. Preferably at least mesa 22.0

**Comment by orklann** [comment]
I just setup non-steam proton game, and here is the log by setting `PROTON_LOG=1 %command%` as the lunch options.

[steam-11195755666436259840.log](https://drive.google.com/file/d/10N6PkdUAiDlyKIRrEh09EEFTID25jBvK/view?usp=sharing)

**Comment by orklann** [comment]
I also set the `DXVK_HUD` to 1, and got the HUB showed up.

![Screenshot from 2022-08-27 10-20-40](https://user-images.githubusercontent.com/5492042/187010572-86bf42ab-6d7e-458b-82a4-a4592fe98ba5.png)

**Comment by orklann** [comment]
I think i was downloading the wrong ` Wine-GE-Proton7-26`, i download the from the official github repo, and i can run Warcraft III reforged, but I get into the liongate screen, then shows nothing with a mouse cursor.

![Screenshot from 2022-08-27 14-00-41](https://user-images.githubusercontent.com/5492042/187017278-ad503249-fb47-4f13-b081-e759b2c10d46.png)

Any help?

**Comment by Blisto91** [comment]
Again there is no need to post many times in a row. Just edit your first post.

Anyway no i don't see anything interesting in your log. You could try turning off esync in lutris ui or steam (`PROTON_NO_ESYNC=1 %command%` for the latter) but i'm not convinced it will help.
This could just be some weirdness with your gpu or driver (version is fine).

**Comment by orklann** [comment]
@Blisto91 I finally make it work to run Warcaft III Reforged on Lutris with `Wine-GE-Proton7-26` in the link you mentions, I just need to wait for the first time the intro video to pass, and after that it crashed, then I just set the window version for `BlizzardBrowser.exe` to windows 7, and also override two dlls for `BlizzardBrowser.exe` to build-in as mentioned in previous post in this thread. All works fine.

Thank you @Blisto91 for your help. :)

**Comment by Blisto91** [comment]
Good to hear you managed to get it working 👍

**Comment by orklann** [comment]
@Blisto91 I checked the Wine-GE github repo, they have a GPU list which they support, and my Intel GPU is in the list, so I just try some different settings, and it works fine. It's not my GPU issue.

**Comment by Blisto91** [comment]
Good to hear.
Alot of Linux gaming can still be a bit janky out of the box. Hopefully it continues to improve so more things just work without tweaking

**Comment by orklann** [comment]
https://github.com/lutris/docs/blob/master/InstallingDrivers.md

Yeah, In this page, it mentioned that `Only Skylake and newer Intel CPUs (processors) offer full Vulkan support`, and my Intel GPU is Skylate. :)

**Comment by SeanMcnabb** [comment]
I'm not sure if this is a dxvk issue or wlroots/wayland issue but anytime I try to run wc3 in a wlroots window manager, I keep getting the following error over and over again and it just hangs at the loading screen. This happens when I set the blizzardbrowser.exe library overrides to wine instead of native

```
06e8:err:winediag:wined3d_dll_init Using the Vulkan renderer.
06e8:fixme:d3d:adapter_vk_init_driver_info Failed to retrieve GPU description for device "AMD Radeon RX 6700 XT (RADV NAVI22)" 1002:73df.
06e8:fixme:kernelbase:AppPolicyGetProcessTerminationMethod FFFFFFFFFFFFFFFA, 000000000011FD70
```

If I keep the overrides set to native, it will result in the following over and over again. 
```07c8:fixme:kernelbase:AppPolicyGetProcessTerminationMethod FFFFFFFFFFFFFFFA, 000000000011FD70```

Everything is working properly though in x window managers, but I would like to get this working in wayland if possible. Anyone have any thoughts?
